Sources/TuistLoader/ProjectDescriptionHelpers/ProjectDescriptionHelpersBuilder.swift

```diff
@@ -41,52 +47,89 @@ final class ProjectDescriptionHelpersBuilder: ProjectDescriptionHelpersBuilding
-        let prefixHash = projectDescriptionHelpersHasher.prefixHash(helpersDirectory: helpersDirectory)
+    private func buildDefaultHelpers(
+        in path: AbsolutePath,
+        projectDescriptionSearchPaths: ProjectDescriptionSearchPaths
+    ) throws -> ProjectDescriptionHelpersModule? {
+        guard let tuistHelpersDirectory = helpersDirectoryLocator.locate(at: path) else { return nil }
+        return try buildHelpers(
+            name: "ProjectDescriptionHelpers",
+            in: tuistHelpersDirectory,
+            projectDescriptionSearchPaths: projectDescriptionSearchPaths
+        )
+    }
+
+    private func buildHelpers(
+        name: String,
+        in path: AbsolutePath,
+        projectDescriptionSearchPaths: ProjectDescriptionSearchPaths
+    ) throws -> ProjectDescriptionHelpersModule {
+        if let cachedModule = builtHelpers[path] { return cachedModule }
+
+        let hash = try projectDescriptionHelpersHasher.hash(helpersDirectory: path)
+        let prefixHash = projectDescriptionHelpersHasher.prefixHash(helpersDirectory: path)
 
-        // Get paths
         let helpersCachePath = cacheDirectory.appending(component: prefixHash)
         let helpersModuleCachePath = helpersCachePath.appending(component: hash)
-        let dylibName = "libProjectDescriptionHelpers.dylib"
+        let dylibName = "lib\(name).dylib"
         let modulePath = helpersModuleCachePath.appending(component: dylibName)
+        let projectDescriptionHelpersModule = ProjectDescriptionHelpersModule(name: name, path: modulePath)
 
-        builtHelpers[helpersDirectory] = modulePath
+        builtHelpers[path] = projectDescriptionHelpersModule
 
         if FileHandler.shared.exists(helpersModuleCachePath) {
-            return modulePath
+            return projectDescriptionHelpersModule
         }
 
         // If the same helpers directory has been previously compiled
         // we delete it before compiling the new changes.
         if FileHandler.shared.exists(helpersCachePath) {
             try FileHandler.shared.delete(helpersCachePath)
         }
+
         try FileHandler.shared.createFolder(helpersModuleCachePath)
 
-        let command = self.command(outputDirectory: helpersModuleCachePath,
-                                   helpersDirectory: helpersDirectory,
-                                   projectDescriptionSearchPaths: projectDescriptionSearchPaths)
+        let command = createCommand(
+            moduleName: name,
+            directory: path,
+            outputDirectory: helpersModuleCachePath,
+            projectDescriptionSearchPaths: projectDescriptionSearchPaths
+        )
         try System.shared.runAndPrint(command, verbose: false, environment: Environment.shared.manifestLoadingVariables)
 
-        return modulePath
+        return projectDescriptionHelpersModule
     }
 
-    // MARK: - Fileprivate
+    private func createCommand(
```

Comment:
I am wondering (I hope this is a good part where to ask that) - are you able to import custom project description helpers in your default ones? Eg. there is a nifty method in a remote helper, but I'd like to build on it in my own `ProjectDescriptionHelpers`, is that possible? From what I could gather, I don't see it, but maybe I am missing something. I believe this is gonna be quite a common scenario, so it'd be great to support this from the start.

I'd also add/modify a fixture to catch this scenario.

Reply:
You are correct that as it stands we cannot import plugins from the default helpers. I originally did not add this support because this way we don't need to worry about cyclic dependency between Plugins and `ProjectDescriptionHelpers`. 

However, if we want to be able to import plugin helpers into the "default" helpers we will need to remove the ability to import the "default" helpers in plugins. I think this makes sense, so I'll be doing that now.

Reply:
Yes, I think it makes sense not to need `ProjectDescriptionHelpers` in `Plugin.swift` 👍